atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -787,23 +827,27 @@ public void stopReadingEarlyWhenEncounteringEntryKnownToBeCommittedAfterSweepTs(
 
         // first iteration reads all before giving up
         sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
-        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(4 + writesInDedicated);
+        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(
+                4 + writesInDedicated + (readBatchSize > 1 ? 1 : 0));
 
         // we read one entry and give up
         sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
-        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(4 + writesInDedicated + 1);
+        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(
+                4 + writesInDedicated + 1 + (readBatchSize > 1 ? 1 : 0));
 
         immutableTs = 170;
 
         // we read one good entry and then a reference to bad entries and give up
         sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
-        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(4 + writesInDedicated + 3);
+        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(
+                4 + writesInDedicated + 3 + (readBatchSize > 1 ? 2 : 0));
 
         immutableTs = 250;
 
         // we now read all to the end
         sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
-        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(4 + writesInDedicated + 3 + writesInDedicated + 2);
+        assertThat(metricsManager).hasEntriesReadConservativeEqualTo(
+                4 + writesInDedicated + 3 + writesInDedicated + 2 + + (readBatchSize > 1 ? 2 : 0));
```

Comment:
`+ +`

Reply:
doh! Amazingly it works because of unary operators